client/src/components/FileExplorer.tsx

```diff
@@ -235,30 +314,34 @@ function FilesTreeView(props) {
   return <div className="tree-container">{tree}</div>;
 }
 
-/**
- * Generic files tree generator.
- * Some things are left to do to make it more generic.
- *
- * @param {*} props.files - This is a list of files with atLocation containing the file path (this is optional)
- * @param {*} props.filesTree - This is the already built fileTree (optional)
- * @param {*} props.foldersOpenOnLoad - Number of folders that should appear open already when displaying the tree
- * @param {*} props.lineageUrl - Should be replaced for URL, this is the link for the file (when clicked) (optional)
- * @param {*} props.insideProject - Boolean to be set true if the display is inside a project
- */
-function FileExplorer(props) {
-  const [filesTree, setFilesTree] = useState(undefined);
+type FileExplorerProps = {
+  files?: JsonObj[]; // This is a list of files with atLocation containing the file path (this is optional)
```

Comment:
These should be jsDoc comments, e.g. they will be displayed by vscode when hovering the prop.
```suggestion
  /** List of files with atLocation containing the file path (this is optional) */
  files?: JsonObj[];
```